resource-managers/mesos/src/main/scala/org/apache/spark/executor/MesosExecutorBackend.scala

```diff
@@ -104,7 +104,8 @@ private[spark] class MesosExecutorBackend
       logError("Received KillTask but executor was null")
     } else {
       // TODO: Determine the 'interruptOnCancel' property set for the given job.
-      executor.killTask(t.getValue.toLong, interruptThread = false)
+      executor.killTask(
+        t.getValue.toLong, interruptThread = false, reason = "killed intentionally")
```

Comment:
can you make this say "killed by Mesos" instead? (assuming this is correct?) "intentionally" doesn't provide much value since it's always intentional